### ISSUE TITLE
Deleted or updated shared entity data is not properly updated in BOXED_type.

### ISSUE THREAD

**Issue by shaper** [PolymerLabs/arcs#890]
Repro'd just now as:

- Have Walt open a new 'Write posts...' arc. Write two messages. Mark the arc as shared w/ friends.
- Have Scott open a new 'Show posts...' arc to view aggregated posts. You'll see the two messages from Walt.
- Have Walt delete one of the posts. Note that the deleted post still shows in Scott's arc. The logging seems to show that FireBase triggers an update, and the `BOXED_posts` are updated to include the one post still remaining from Walt, but that seems purely additive/update; that is, we never remove the deleted one from the existing `BOXED_posts` handle.

Log excerpt on Scott's arc side after Walt deletes the post below. The deleted post had message as 'Item One', and the remaining post had message as 'Item Two':

```
READING friend's [Walt] shared handles
remote-friends-shared-handles.js:151 RemoteFriendsSHs added friend's shared data to box [BOXED_blog_metadata] [{…}]
remote-friends-shared-handles.js:151 RemoteFriendsSHs added friend's shared data to box [BOXED_posts] 
[{…}]
0
:
id
:
"!987396311182881:demo-g51iznrav0rbb0uf:0:inner:5"
rawData
:
{author: "-L-8Ie9J417_6rBeMseD", createdTimestamp: 1520373548292, message: "Item Two.", owner: "-L-8Ie9J417_6rBeMseD"}
__proto__
:
Object
length
:
1
__proto__
:
Array(0)
````

And viewing the handle in Handle Explorer shows both the deleted item and the updated item above:

```
values
0
author
-L-8Ie9J417_6rBeMseD
createdTimestamp
1520288410816
message
An initial message.
owner
-L-8Ie9J417_6rBeMseD
1
author
-L-8Ie9J417_6rBeMseD
createdTimestamp
1520289667098
message
One more test.
owner
-L-8Ie9J417_6rBeMseD
2
author
-L-8Ie9J417_6rBeMseD
createdTimestamp
1520373542907
message
Item One.
owner
-L-8Ie9J417_6rBeMseD
3
author
-L-8Ie9J417_6rBeMseD
createdTimestamp
1520373548292
message
Item Two.
owner
-L-8Ie9J417_6rBeMseD
````

**Comment by shaper** [comment]
@seefeldb this is the issue tracking the duplicate entries we saw today in the aggregated social feed demo.

I find that if I create a Words game, it creates an initial `Stats` record which is boxed and shows in the feed. If I then make a move, that same `Stats` record is updated by the Words game.

However, a separate social feed arc kept open throughout will see `BOXED_stats` updated initially so that it contains the first original `Stats` data and, subsequently, a new, second `Stats` data record even though there should just be one whose content was updated.
